OpenRA.Mods.Common/Traits/Render/DrawLineToTarget.cs

```diff
@@ -66,37 +65,40 @@ IEnumerable<IRenderable> IRenderAboveShroudWhenSelected.RenderAboveShroud(Actor
 		{
 			var force = Game.GetModifierKeys().HasModifier(Modifiers.Alt);
 			if ((lifetime <= 0 || --lifetime <= 0) && !force)
-				yield break;
+				return new IRenderable[0];
 
 			if (!(force || Game.Settings.Game.DrawTargetLine))
-				yield break;
+				return new IRenderable[0];
 
-			if (targets == null || targets.Count == 0)
-				yield break;
+			if (self.GetCurrentActivity() == null)
+				return new IRenderable[0];
 
-			foreach (var target in targets)
-			{
-				if (target.Type == TargetType.Invalid)
-					continue;
+			if (self.GetCurrentActivity().GetTargets(self).Count() <= 0)
```

Comment:
You also do GetTargets(self) 2 times,put that into a variable too as it can become expensive depending on the implementation of the activity.